src/Features/AutoSubmit.cpp

```diff
@@ -421,6 +423,108 @@ CON_COMMAND_F(sar_challenge_autosubmit_reload_api_key, "sar_challenge_autosubmit
 	loadApiKey(true);
 }
 
+size_t writeCallback(void* contents, size_t size, size_t nmemb, std::string* buffer)
+{
+	size_t realsize = size * nmemb;
+	buffer->append((char *)contents, realsize);
+	return realsize;
+}
+
+void retrieveMtriggers(int rank, std::string map_name)
+{
+	bool keyFound = false;
+	if (map_name.empty())
+		return THREAD_PRINT("Not playing a map.\n");
+
+	for (const auto& pair : g_map_ids)
+	{
+		if (pair.first == map_name)
+		{
+			keyFound = true;
+			break;
+		}
+	}
+
+	if (rank < 1)
+		return THREAD_PRINT("Invalid rank.\n");
+
+	if (keyFound) {
+		CURL *curl;
+		CURLcode res;
+		std::string buffer;
+		curl = curl_easy_init();
+		if (curl) {
+			std::string apiCall = "/v1/mtriggers/search?game_dir=portal2&map_name=" + map_name + "&board_rank=" + std::to_string(rank);
+			curl_easy_setopt(curl, CURLOPT_URL, API_BASE_AUTORENDER + apiCall);
+			curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeCallback);
+			curl_easy_setopt(curl, CURLOPT_WRITEDATA, &buffer);
+			res = curl_easy_perform(curl);
+
+			if (res != CURLE_OK)
+				return THREAD_PRINT("Failed to retrieve.\n");
```

Comment:
Early returns should call `curl_easy_cleanup` or this will leak the cURL handle.